src/NBitcoin/Network.cs

```diff
@@ -336,7 +363,8 @@ internal static Network Register(NetworkBuilder builder)
             network.DefaultPort = builder.Port;
             network.RPCPort = builder.RPCPort;
             network.genesis = builder.Genesis;
-            network.consensus.HashGenesisBlock = network.genesis.GetHash();
+            if(network.consensus != null && network.genesis != null)
```

Comment:
This `if` is not correct both `consensus` and `genesis` should be set so we need to throw if they are not set.